HealthMonitoring.Management.Core/EndpointMetricsForwarderCoordinator.cs

```diff
@@ -23,13 +21,24 @@ public EndpointMetricsForwarderCoordinator(IEnumerable<IEndpointMetricsForwarder
                     Logger.WarnFormat("Forwarder with type {0} already exists and it is not going to be registered", type);
             }
         }
-        
-        public void HandleMetricsForwarding(Guid endpointId, EndpointHealth stats)
+
+        public void HandleMetricsForwarding(Endpoint endpoint)

```

Comment:
I would probably pass here endpoint identity, metadata and health separately.
The reason is that endpoint object may mutate in the meantime if other thread update health, metadata or even destroy object.